frontend/src/utils/index.ts

```diff
@@ -178,3 +179,13 @@ export const getColumnFromName = (
   columns: AdvancedQueryColumn[],
 ): AdvancedQueryColumn | undefined =>
   columns.find((col) => col.name === name) as AdvancedQueryColumn;
+
+export const getPreference = (token: string) => {
+  return window.fetch(api.routes.USERPREFERENCE, {
+    method: 'GET',
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `token ${token}`,
+    },
+  });
```

Comment:
Would be best if you returned the actual JSON here i.e `response.json()` ... abstract as much within this function as you can so that usages of it don't have to do as much ... also, lets see you can do it with `async ... await`.